### PR TITLE
restructured install documentation

### PR DESCRIPTION
**Pull request by inteon**
Instead of Kubernetes vs OpenShift, the installation categories should be based on the installation tools.

### CODE REVIEW

File: content/en/docs/installation/helm.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+> upgrade to a supported version before installing cert-manager.
+
+> **Warning**: You should not install multiple instances of cert-manager on a single
+> cluster. This will lead to undefined behavior and you may be banned from
+> providers such as Let's Encrypt.
+
+## Installing with Helm
+
+> **Note**: cert-manager should never be embedded as a sub-chart into other Helm charts.
+> cert-manager manages non-namespaced resources in your cluster and should only be installed once.
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:19]: suggestion: I don't think this needs to be formatted as a note, and we could add a little more intro text:

```suggestion
cert-manager provides Helm charts as a first-class method of installation on both Kubernetes and OpenShift.

Be sure never to embed cert-manager as a sub-chart of other Helm charts; cert-manager manages
non-namespaced resources in your cluster and care must be taken to ensure that it is installed exactly once.
```

File: content/en/docs/installation/helm.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+### Prerequisites
+
+- Helm v3 installed
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:23]: suggestion: as mentioned above

```suggestion
- Helm version 3 or later
- A Kubernetes or OpenShift cluster running a [supported version](link/to/supported-versions/page/here)
- cert-manager not already installed on the cluster
```

File: content/en/docs/installation/helm.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+### Steps
+
+In order to install the Helm chart, you must follow these steps:
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:27]: suggestion: we can remove this text entirely, it's not adding anything

```suggestion
```

File: content/en/docs/installation/helm.md [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+#### 1. Add the Jetstack Helm repository:
+
+> **Warning**: It is important that this repository is used to install
+> cert-manager. The version residing in the [Helm stable repository](https://github.com/helm/charts/tree/master/stable/cert-manager) is
+> *deprecated* and should *NOT* be used.
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:33]: suggestion: I don't think we should link to a deprecated repository; we should just encourage people to use our repo. I also don't think this needs to be highlighted as a warning.

```suggestion
This repository is the only supported source of cert-manager charts. There are some other mirrors and copies across the internet, but those are entirely unofficial and could present a security risk.

Notably, the "Helm stable repository" version of cert-manager is deprecated and should not be used.
```

File: content/en/docs/installation/helm.md [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+$ helm repo update
+```
+
+#### 3. Install `CustomResourceDefinitions`
+
+cert-manager requires a number of CRD resources to be installed into your
+cluster as part of installation.
+
+This can either be done manually, using `kubectl`, or using the `installCRDs`
+option when installing the Helm chart.
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:51]: suggestion: this is weirdly phrased and too wordy, I think

```suggestion
cert-manager requires a number of CRD resources, which can  be installed manually using `kubectl`,
or using the `installCRDs` option when installing the Helm chart.
```

File: content/en/docs/installation/helm.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+> **Note**: If you're using a `helm` version based on Kubernetes `v1.18` or below (Helm `v3.2`) `installCRDs` will not work with cert-manager `v0.16`.
+> For more info see the [v0.16 upgrade notes](../upgrading/upgrading-0.15-0.16/#helm)
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:54]: suggestion: this can be moved under the option 2 heading below, since it's only relevant to that option

File: content/en/docs/installation/helm.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+##### Option 1: installing CRDs with `kubectl`
+
+Install the `CustomResourceDefinition` resources using `kubectl`:
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:58]: suggestion: this text doesn't provide any new information and I think it can be removed

```suggestion
```

File: content/en/docs/installation/helm.md [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+$ kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v1.4.0/cert-manager.crds.yaml
+```
+
+##### Option 2: install CRDs as part of the Helm release
+
+To automatically install and manage the CRDs as part of your Helm release, you
+must add the `--set installCRDs=true` flag to your Helm installation command.
+
+Uncomment the relevant line in the next steps to enable this.
+
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:70]: suggestion: add the block about older versions of Helm here


```suggestion

Note that if you're using a `helm` version based on Kubernetes `v1.18` or below (Helm `v3.2`), `installCRDs` will not work with cert-manager `v0.16`. See the [v0.16 upgrade notes](../upgrading/upgrading-0.15-0.16/#helm) for more details.

```

File: content/en/docs/installation/helm.md [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+```bash
+$ helm install \
+  cert-manager jetstack/cert-manager \
+  --namespace cert-manager \
+  --create-namespace \
+  --version v1.4.0 \
+  # --set installCRDs=true
+```
+
+On the [cert-manager's ArtifactHub page](https://artifacthub.io/packages/helm/cert-manager/cert-manager) a full list of the available **Helm values** can be found.
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:84]: suggestion: rewording

```suggestion
A full list of available Helm values is on [cert-manager's ArtifactHub page](https://artifacthub.io/packages/helm/cert-manager/cert-manager).
```

File: content/en/docs/installation/helm.md [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+  cert-manager jetstack/cert-manager \
+  --namespace cert-manager \
+  --create-namespace \
+  --version v1.4.0 \
+  --set prometheus.enabled=false \  # Example: disabling prometheus using a Helm parameter
+  --set webhook.timeoutSeconds=4s   # Example: changing the wehbook timeout using a Helm parameter
+```
+
+Once you have deployed cert-manager, you can verify the installation
+[here](../verify/).
```
Comment (SgtCoDFish) [inline content/en/docs/installation/helm.md:99]: suggestion: make the verb the link, rather than adding a "here"

```suggestion
Once you have deployed cert-manager, you can [verify](../verify/) the installation.
```

File: content/en/docs/installation/kubectl.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
+---
+title: "Kubectl apply"
+linkTitle: "Kubectl apply"
+weight: 20
+type: "docs"
+---
+
+> Note: From cert-manager `v1.2.0` onward, the minimum supported version of
+> Kubernetes is `v1.16.0`. Users still running Kubernetes `v1.15` or below should
+> upgrade to a supported version before installing cert-manager.
```
Comment (SgtCoDFish) [inline content/en/docs/installation/kubectl.md:10]: note: as with the helm doc, I think there are just too many notes here, but I've made enough suggestions already with the helm doc, so I'm not going to go through here and suggest changes :joy:  We should revisit and reword a lot of this but I don't think we need to do that in this PR.

File: content/en/docs/installation/_index.md [hunk 12]
```diff
@@ -6,6 +6,6 @@ type: "docs"
 ---
```
Comment (SgtCoDFish) [inline content/en/docs/installation/_index.md:7]: suggestion: I think the new "scenarios" page could actually just be placed here, on the index page for the `Installation` section. I'm not sure what we gain by having "scenarios" on a different page.

File: content/en/docs/installation/scenarios.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
+---
+title: "Scenarios"
+linkTitle: "Scenarios"
+weight: 10
+type: "docs"
+---
```
Comment (SgtCoDFish) [inline content/en/docs/installation/scenarios.md:6]: suggestion: as noted before, I think this doesn't need to be a separate page; it could just go into the `_index` for `Installation`, and the index page there could be the one-stop-shop for describing different types of installation

File: content/en/docs/installation/verify.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+cert-manager-cainjector-577f6d9fd7-tr77l   1/1     Running   0          2m
+cert-manager-webhook-787858fcdb-nlzsq      1/1     Running   0          2m
+```
+
+You should see the `cert-manager`, `cert-manager-cainjector`, and
+`cert-manager-webhook` pod in a `Running` state.
+It may take a minute or so for the TLS assets required for the webhook to
+function to be provisioned. This may cause the webhook to take a while longer
+to start for the first time than other pods. If you experience problems, please
+check the [FAQ guide](../../faq/).
```
Comment (SgtCoDFish) [inline content/en/docs/installation/verify.md:25]: suggestion: a slight rewording

```suggestion
You should see the `cert-manager`, `cert-manager-cainjector`, and
`cert-manager-webhook` pods in a `Running` state. The webhook might take a
little longer to successfully provision than the others.

If you experience problems, first check the [FAQ](../../faq/).
```

File: content/en/docs/installation/verify.md [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+  ----    ------      ----  ----          -------
+  Normal  CertIssued  4s    cert-manager  Certificate issued successfully
+```
+
+Clean up the test resources.
+```bash
+$ kubectl delete -f test-resources.yaml
+```
+
+If all the above steps have completed without error, you are good to go!
```
Comment (SgtCoDFish) [inline content/en/docs/installation/verify.md:95]: nitpick: this is a real nitpick :joy:  if we're using more informal language like "good to go" (which I think is good here!) I think this reads better using a contraction for "you are"

```suggestion
If all the above steps have completed without error, you're good to go!
```

File: content/en/docs/installation/verify.md [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+
+Optionally the whole verification flow is automated by running tool maintained by the community [cert-manager-verifier](https://github.com/alenkacz/cert-manager-verifier).
```
Comment (SgtCoDFish) [inline content/en/docs/installation/verify.md:97]: suggestion: this sentence might be better higher up in this document; it would be annoying to do several manual steps to test your installation only to find out at the end of the document that there's a tool which can do it for you!

also, I'd reword slightly:

```suggestion
To automatically check if cert-manager is correctly configured, you can run the community-maintained [cert-manager-verifier](https://github.com/alenkacz/cert-manager-verifier) tool.
```

File: content/en/docs/installation/verify.md [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+
+If you experience problems, please check the
+[FAQ](../../faq/).
```
Comment (SgtCoDFish) [inline content/en/docs/installation/verify.md:100]: suggestion: We mention the FAQ above, I don't think we need to mention it again here

```suggestion
```

**Review by SgtCoDFish — COMMENTED** [review]
this definitely looks like an improvement!

I know I've made a lot of comments here, but I think mostly it's stuff that was copy pasted from old docs. I like the direction this is going!

File: content/en/docs/installation/verify.md (hunk 16 above)
Comment (inteon) [reply content/en/docs/installation/verify.md]: Indeed, but the "kubectl cert-manager check api" command will replace this tool anyway, I'll revise this verify.md doc in a later PR.

File: content/en/docs/installation/kubectl.md [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+> 'permission denied' error when creating some of these resources. This is a
+> nuance of the way GKE handles RBAC and IAM permissions, and as such you should
+> 'elevate' your own privileges to that of a 'cluster-admin' **before** running
+> the above command. If you have already run the above command, you should run
+> them again after elevating your permissions:
+```bash
+$ kubectl create clusterrolebinding cluster-admin-binding \
+    --clusterrole=cluster-admin \
+    --user=$(gcloud config get-value core/account)
+```
```
Comment (SgtCoDFish) [inline content/en/docs/installation/kubectl.md:38]: suggestion: it doesn't feel quite right having a cloud-provider-specific suggestion here in this block. Could we maybe add a new subheading under the line `Once you have deployed cert-manager, you can verify the installation [here](../verify/).` below?

```suggestion
### Permissions Errors on Google Kubernetes Engine

When running on , you might encounter a 'permission denied' error when creating some
of the required resources. This is a nuance of the way GKE handles RBAC and IAM permissions,
and as such you might need to elevate your own privileges to that of a "cluster-admin" **before**
running `kubectl apply`.

If you have already run `kubectl apply`, you should run it again after elevating your permissions:

```bash
$ kubectl create clusterrolebinding cluster-admin-binding \
    --clusterrole=cluster-admin \
    --user=$(gcloud config get-value core/account)
\``` # 🔥 should just be three backticks but it's hard to type three backticks inside a suggestion block 😂
```

File: content/en/docs/installation/operator-lifecycle-manager.md [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+type: "docs"
+---
+
+## Installation managed by OLM
+
+### Prerequisites
+
+- A Kubernetes or OpenShift cluster running a [supported version](../supported-releases/)
+- cert-manager not already installed on the cluster
+- [Prerequisites specific to Cloud provider](../compatibility/)
```
Comment (SgtCoDFish) [inline content/en/docs/installation/operator-lifecycle-manager.md:14]: suggestion: minor wording change

```suggestion
- [Prerequisites specific to your cloud provider](../compatibility/)
```

File: content/en/docs/installation/operator-lifecycle-manager.md [hunk 20]
```diff
@@ -0,0 +1,102 @@
+---
+title: "Operator Lifecycle Manager"
```
Comment (SgtCoDFish) [inline content/en/docs/installation/operator-lifecycle-manager.md:2]: note: I'm not at all familiar with OLM, so I haven't reviewed anything technical in this page - just the wording

File: content/en/docs/installation/verify.md [hunk 21]
```diff
@@ -0,0 +1,105 @@
+---
+title: "Verifying the installation"
```
Comment (SgtCoDFish) [inline content/en/docs/installation/verify.md:2]: suggestion: as above, title casing

```suggestion
title: "Verifying the Installation"
```

**Review by SgtCoDFish — CHANGES_REQUESTED** [review]
good improvements! just a few more little bits; I'm sorry to be so thorough! :sweat_smile:

File: content/en/docs/installation/kubectl.md [hunk 22]
```diff
@@ -0,0 +1,92 @@
+---
+title: "Kubectl apply"
+linkTitle: "Kubectl apply"
```
[the remark this answers is not in the document]
Comment (inteon) [reply content/en/docs/installation/kubectl.md]: I think both look a bit weird on the website ("Kubectl Apply" & "kubectl apply"), https://cert-manager.io/docs/usage/kubectl-plugin/ also says `Kubectl plugin`, and there are some other places on the website where the casing is different.
So, personally, I would prefer to not change it, please let me know if you don't agree.

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
